src/main/java/org/spongepowered/api/service/placeholder/PlaceholderText.java

```diff
@@ -0,0 +1,176 @@
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.text.TextRepresentable;
+import org.spongepowered.api.text.channel.MessageReceiver;
+import org.spongepowered.api.util.ResettableBuilder;
+
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Supplier;
+
+/**
+ * A {@link TextRepresentable} that can be used in {@link Text} building methods
+ * that represents a placeholder in text.
+ *
+ * <p>A {@link PlaceholderText} is the collection of a {@link PlaceholderParser}
+ * along with contextual data, enabling its use in a {@link Text} object.</p>
+ *
+ * <p>While such placeholders will generally be built from tokenised strings,
+ * these objects make no assumption about the format of text templating. Such a
+ * system can therefore be used by other templating engines without conforming
+ * to a particular standard.</p>
+ */
+public interface PlaceholderText extends TextRepresentable {
+
+    /**
+     * Gets a builder for creating {@link PlaceholderText}.
+     *
+     * @return A {@link Builder}
+     */
+    static PlaceholderText.Builder builder() {
+        return Sponge.getServiceManager().provideUnchecked(PlaceholderService.class).placeholderBuilder();
+    }
+
+    /**
+     * Gets the {@link PlaceholderParser} that handles this
+     * placeholder.
+     *
+     * @return The {@link PlaceholderParser}
+     */
+    PlaceholderParser getParser();
+
+    /**
+     * If provided, the {@link MessageReceiver} which to pull information
+     * from when building the placeholder text.
+     *
+     * <p>Examples of how this might affect a placeholder are:</p>
+     *
+     * <ul>
+     *     <li>
+     *         For a "name" placeholder that prints out the source's name,
+     *         the name would be selected from this source.
+     *     </li>
+     *     <li>
+     *         For a "current world" placeholder that returns a player's
+     *         monetary current world, this would pull the balance from the
+     *         player.
+     *     </li>
+     * </ul>
+     *
+     * <p>It is important to note that the associated source does not
+     * necessarily have to be the sender/invoker of a message, nor does it
+     * have to be the recipient. The source is selected by the context of
+     * builder. It is up to plugins that use such placeholders to be aware
+     * of the context of which the placeholder is used.</p>
+     *
+     * <p>If an invalid {@link MessageReceiver} is provided for the context
+     * of the placeholder, then the associated {@link PlaceholderParser} must
+     * return a {@link Text#EMPTY}.</p>
+     *
+     * @return The associated {@link MessageReceiver}, if any.
+     */
+    Optional<MessageReceiver> getAssociatedReceiver();
+
+    /**
+     * The variable string passed to this token to provide contextual
+     * information.
+     *
+     * @return The argument, if any
+     */
+    Optional<String> getArgument();
```

Comment:
```suggestion
    Optional<String> getArguments();
```
Since the argument in PlaceholderService was modified from argument to argumentString, it'd likely be best to make that modification here as well.

Reply:
Gone for `getArgumentString` instead.